gdnative-derive/src/methods.rs

```diff
@@ -229,9 +236,66 @@ fn impl_gdnative_expose(ast: ItemImpl) -> (ItemImpl, ClassMethodExport) {
-                                    let unexpected = last.ident.to_string();
-                                    let msg =
-                                        format!("unknown option for export: `{}`", unexpected);
+                                    let path = last.ident.to_string();
+
+                                    // Match rpc mode
+                                    match path.as_str() {
+                                        "rpc" => {
+                                            let value = if let syn::Lit::Str(lit_str) = lit {
+                                                lit_str.value()
+                                            } else {
+                                                errors.push(syn::Error::new(
+                                                    last.span(),
+                                                    "unexpected type for rpc value, expected Str",
+                                                ));
+                                                return false;
+                                            };
+
+                                            match value.as_str() {
+                                                "remote" => {
+                                                    rpc = "remote";
+                                                    return false;
+                                                }
+                                                "remotesync" => {
+                                                    rpc = "remotesync";
+                                                    return false;
+                                                }
+                                                "master" => {
+                                                    rpc = "master";
+                                                    return false;
+                                                }
+                                                "puppet" => {
+                                                    rpc = "puppet";
+                                                    return false;
+                                                }
+                                                "disabled" => {
+                                                    rpc = "disabled";
+                                                    return false;
+                                                }
+                                                "mastersync" => {
+                                                    rpc = "mastersync";
+                                                    return false;
+                                                }
+                                                "puppetsync" => {
+                                                    rpc = "puppetsync";
+                                                    return false;
+                                                }
+                                                _ => {
```

Comment:
Underscores should be inserted before `sync` for the `*sync` options, i.e. `master_sync` instead of `mastersync`, as per the rules for `snake_case`.

Reply:
Gotcha, so as a rule of thumb we always prefer snake_case to the Godot documentation?

Reply:
We prefer Rust naming conventions in the bindings, and here the convention is to use `snake_case`.

Reply:
Got it thanks!